### DISCUSSION TITLE
[Tuist] Modularization / Dependency 설계

### DISCUSSION THREAD

**Discussion by chaneeii** [Nexters/www-iOS#6] [General]
## Modularization / Dependency 설계 (1차)

- 아직 Feature 가 나뉠 볼륨이 아니라 타겟은 Application 한개로 운용
- DIContainer 따로 모듈을 만들어서 관리하면 좋을듯! -> 아직 기능이 다 안나와서 쫌더 고민필요!

개인적인 욕심으로 한다면 1안! 어차피 트위스트 시작한 이상 설계만 잘 해두면 1안이나 2안이나 노력의 정도 / 비용 측면에서는 비슷하게 들어갈 것으로 예상 
=> 일단 프로젝트는 둘다 잘 돌아감
=> 단, 종속적인 순환 관계 가 생길수 있다는 문제가 있는데 지금까지 나온 기능으로 1안으로 설계했을때는 미친 실수를 하지 않는 이상 없을 것 같음 (이라고 말하지만 언제나 개발은 제맘대로 안됩니다.) 

> 사실 가볍게 쓰려했는데 설계하다 보니깐 그래도 의미가 있어야지 하면서 쪼개고쪼개다 커졌어요 😂

### 다이어그램 보는법 참고
<img src="https://user-images.githubusercontent.com/63157395/213776677-9a6a3a52-39c3-4049-94a5-7364264d9f10.png" width="500">

### 1안
- 현재나온 기능 및 feature에서 가장 이상적인 형태가 아닐까 ??? 싶습니다 (최소 제 두뇌가 동작하는 범위에서)
<img src="https://user-images.githubusercontent.com/63157395/213775197-5bcbf901-be7b-44dd-9b5a-42d94cf60503.png" width="500">

### 2안 
1안이 너무 헤비하다면 2안이 있으나,, 1안이 너무 이상적임 

이 친구들을 이렇게 다 합쳐서 Application 으로 넣기 
<img src="https://user-images.githubusercontent.com/63157395/213775549-4c1920f1-8ec6-4913-8d72-8a21820dae3b.png" width="500">

=> 이러면 데이터레이어 구분 및 충돌방지목적정도..
<img src="https://user-images.githubusercontent.com/63157395/213775838-a2011378-1199-4b48-a1b7-90811f28ef57.png" width="500">

### ref
- https://docs.tuist.io/commands/graph/
- https://docs.tuist.io/building-at-scale/microfeatures
- https://minsone.github.io/mac/ios/ios-clean-architecture-with-ribs-reactorkit-and-modularization-using-tuist-1
- https://ios-development.tistory.com/1227
- https://github.com/kyungpyoda/Unsplasher
- [[오강훈] Modular_Architecture_시작하기.pdf](https://github.com/Nexters/www-iOS/files/10469088/Modular_Architecture_.pdf)
- [카카오뱅크iOS모듈화여정.pdf](https://github.com/Nexters/www-iOS/files/10469091/iOS.pdf)

**Comment by chaneeii** [comment]
@kokojong  의 의견이 궁금합니다!

**Comment by kokojong** [comment]
저 노란색으루 한게 하나의 모듈인곤가??

**Reply by chaneeii** [reply]
노란색, 빨간색이요!

**Reply by chaneeii** [reply]
복잡해보이지만 간단한데 내일 만나서 시간되면 설명해줄게요...!
